ci-operator/step-registry/gather/must-gather/gather-must-gather-commands.sh

```diff
@@ -1,8 +1,6 @@
 #!/bin/bash
 
 set -o nounset
-set -o errexit
```

Comment:
This will make the `mkdir` non-fatal as well, and I expect we want to hear about that as a failing step.  [rbhz#1940894][1] should eventually make `must-gather` more robust to the OpenShift API-server outages, and [rhbz#1969598][2] is about fixing the OpenShift API-server outages themselves.  I'm not sure we need additional step changes beyond waiting for those two.  If we do carve out only failing `must-gather`, something like:

```sh
set +e
oc --insecure-skip-tls-verify adm must-gather $MUST_GATHER_IMAGE --dest-dir ${ARTIFACT_DIR}/must-gather > ${ARTIFACT_DIR}/must-gather/must-gather.log
ret="$?"
set -e

[ -f "${ARTIFACT_DIR}/must-gather/event-filter.html" ] && cp "${ARTIFACT_DIR}/must-gather/event-filter.html" "${ARTIFACT_DIR}/event-filter.html"
tar -czC "${ARTIFACT_DIR}/must-gather" -f "${ARTIFACT_DIR}/must-gather.tar.gz" .
rm -rf "${ARTIFACT_DIR}"/must-gather
exit "${ret}"
```
Should work.  But I dunno if it's worth the trouble.

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=1940894
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1969598

Reply:
If mkdir fails you'll hear about it real quick in the next steps, we're not hiding any errors but we are making sure @deads2k doesn't have to torture himself downloading half-collected must-gathers one file at a time.